libgrease/src/crypto/zk_objects.rs

```diff
@@ -197,15 +333,19 @@ pub struct ShardInfo {
 pub fn generate_txc0_nonces<R: CryptoRng + RngCore>(rng: &mut R) -> Comm0PrivateInputs {
     Comm0PrivateInputs {
         random_blinding: GenericScalar::random(rng),
-        a1: GenericPoint::random(rng),
-        r1: GenericPoint::random(rng),
-        r2: GenericPoint::random(rng),
+        a1: GenericScalar::random(rng),
+        r1: GenericScalar::random(rng),
+        r2: GenericScalar::random(rng),
         blinding_dleq: GenericScalar::random(rng),
     }
 }
 
-pub fn random_256_bits<R: CryptoRng + RngCore>(rng: &mut R) -> [u8; 32] {
+pub fn random_251_bits<R: CryptoRng + RngCore>(rng: &mut R) -> [u8; 32] {
```

Comment:
This is used to generate random Generic scalars, which assume all 32 bytes are in range. There;s no downside to using  random_256_bits in a 251 bit context. But always having to MSB 5 bits zero is dangerous.

Reply:
I'm concerned about having a scalar value that is outside the range of the protocol maximum, with is the BJJ order. I think it should work OK on the Rust side since both the BJJ and Ed25519 libraries handle overflow. But the Noir code cannot since that is a serious side-channel attack. My hack-and-slash solution was to just cap the RNG at 251 bits (which is slightly less than BJJ order). I can fix this now but I don't see any problem leaving this as a post-POC implementation detail.